samd/timers.c

```diff
@@ -87,12 +88,17 @@ void tc_reset(Tc* tc) {
     }
 }
 
-void shared_timer_handler(bool is_tc, uint8_t index) {
-    // Add calls to interrupt handlers for specific functionality here.
-    if (is_tc) {
-        pulseout_interrupt_handler(index);
-    }
-}
+void shared_timer_handler(bool is_tc, uint8_t index, bool pulseout) {

```

Comment:
Weren't we going to move this into CircuitPython?

Instead of inferring pulseout, please keep it as state. See here for an example: https://github.com/adafruit/samd-peripherals/blob/master/samd/external_interrupts.c#L40

Reply:
On second thought, with #1109 in mind, I think `timer_handler.c` makes better sense.